validator/sawtooth_validator/state/client_handlers.py

```diff
@@ -1094,3 +1102,10 @@ def _respond(self, request):
         return self._wrap_response(
             endpoint=self._gossip.endpoint,
             peers=sorted(peers, key=lambda peer: peer.endpoint))
+
+
+def client_batch_submit_request_preprocessor(message_content_bytes):
```

Comment:
Is this a full implementation of this preprocessor? It doesn't seem to be storing the parsed protobuf back in the dispatcher.

Reply:
The preprocessors are side effect-free. The result is stored at https://github.com/hyperledger/sawtooth-core/pull/1489/files#diff-5b210e75615ded71391b5b742dbc779eR204

Reply:
I think what is important that I didn't realize, is that `content=future.result(timeout)` in the defined function that gets called when the future resolves.